esp-hal/src/dma/buffers.rs

```diff
@@ -4,25 +4,252 @@ use core::{
+            /// The default external memory burst length.
+            pub const DEFAULT: Self = Self::Size16;
+        }
+
+        impl Default for ExternalBurstSize {
+            fn default() -> Self {
+                Self::DEFAULT
+            }
+        }
+
+        /// Internal memory access burst mode.
+        #[derive(Clone, Copy, PartialEq, Eq, Debug)]
+        #[cfg_attr(feature = "defmt", derive(defmt::Format))]
+        pub enum InternalBurstTransfer {
+            /// Burst mode is disabled.
+            Disabled,
+
+            /// Burst mode is enabled.
+            Enabled,
+        }
+
+        impl InternalBurstTransfer {
+            /// The default internal burst mode configuration.
+            pub const DEFAULT: Self = Self::Disabled;
+        }
+
+        impl Default for InternalBurstTransfer {
+            fn default() -> Self {
+                Self::DEFAULT
+            }
+        }
+
+        /// Burst transfer configuration.
+        #[derive(Clone, Copy, PartialEq, Eq, Debug)]
+        #[cfg_attr(feature = "defmt", derive(defmt::Format))]
+        pub struct BurstConfig {
+            /// Configures the burst size for PSRAM transfers.
+            ///
+            /// Burst mode is always enabled for PSRAM transfers.
+            pub external: ExternalBurstSize,
+
+            /// Enables or disables the burst mode for internal memory transfers.
+            ///
+            /// The burst size is not configurable.
+            pub internal: InternalBurstTransfer,
+        }
+
+        impl BurstConfig {
+            /// The default burst mode configuration.
+            pub const DEFAULT: Self = Self {
+                external: ExternalBurstSize::DEFAULT,
+                internal: InternalBurstTransfer::DEFAULT,
+            };
+        }
+
+        impl Default for BurstConfig {
+            fn default() -> Self {
+                Self::DEFAULT
+            }
+        }
+    } else {
+        /// Burst transfer configuration.
+        #[derive(Clone, Copy, PartialEq, Eq, Debug)]
+        #[cfg_attr(feature = "defmt", derive(defmt::Format))]
+        pub enum BurstConfig {
+            /// Burst mode is disabled.
+            Disabled,
+
+            /// Burst mode is enabled.
+            Enabled,
+        }
+
+        impl BurstConfig {
+            /// The default burst mode configuration.
+            pub const DEFAULT: Self = Self::Disabled;
+        }
+
+        impl Default for BurstConfig {
+            fn default() -> Self {
+                Self::DEFAULT
+            }
+        }
 
-    /// Burst mode is enabled.
-    Enabled,
+        type InternalBurstTransfer = BurstConfig;
+    }
 }
 
-impl BurstConfig {
+#[cfg(psram_dma)]
+impl ExternalBurstSize {
+    const fn min_psram_alignment(self, direction: TransferDirection) -> usize {
+        // S2: Specifically, size and buffer address pointer in receive descriptors
+        // should be 16-byte, 32-byte or 64-byte aligned. For data frame whose
```

Comment:
should?

Reply:
Yeah this is a TRM copy-paste, I'd not change their wording even though should isn't used to specify a requirement.